Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1

```diff
@@ -0,0 +1,795 @@
+﻿<###
+# Overview:
+    This script is used to configure WAF Policy on all endpoints of Front Door CDN(s) in a Subscription.
+
+# Control ID:
+    Azure_FrontDoor_CDNProfile_NetSec_Enable_WAF_Configuration
+
+# Display Name:
+   Front Door CDN should have Web Application Firewall configured
+
+# Prerequisites:
+    1. Contributor or higher privileges on the Front Door CDNs in a Subscription.
+    2. Must be connected to Azure with an authenticated account.
+
+# Steps performed by the script:
+    To remediate:
+        1. Validate and install the modules required to run the script.
+        2. Get the list of all Front Door CDNs Endpoints in a Subscription that do not have WAF Policy Configured
+        3. Back up details of Front Door CDN Endpoint(s) that are to be remediated.
+        4. Configure WAF for all endpoints in the Front Door.
+
+    To roll back:
+        1. Validate and install the modules required to run the script.
+        2. Get the list of Front Door' Endpoint(s) in a Subscription, the changes made to which previously, are to be rolled back.
+        3. Revert by removing WAF Policies from endpoints in all the Front Door.
+
+# Instructions to execute the script:
+    To remediate:
+        1. Download the script.
+        2. Load the script in a PowerShell session. Refer https://aka.ms/AzTS-docs/RemediationscriptExcSteps to know more about loading the script.
+        3. Execute the script to configure WAF Policy on all endpoints of Front Door CDNs in a Subscription. Refer `Examples`, below.
+
+    To roll back:
+        1. Download the script.
+        2. Load the script in a PowerShell session. Refer https://aka.ms/AzTS-docs/RemediationscriptExcSteps to know more about loading the script.
+        3. Execute the script to remove the configured WAF Policy on all endpoints of Front Door CDNs in a Subscription. Refer `Examples`, below.
```

Comment:
We should say roll back the configuration to previous state as we dont remove it for all the CDNs in subscription.

Reply:
this is done